telegram/base.py

```diff
@@ -80,3 +80,25 @@ def to_dict(self):
                     data[key] = value
 
         return data
+
+    def _get_id(self):
+        for x in ['id', 'file_id', 'message_id', 'result_id', 'update_id']:
+            if hasattr(self, x):
+                return self.__getattribute__(x)
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
```

Comment:
This is one of the few places where I prefer comparing the class, instead of using `isinstance`. The reason is that multiple levels of inheritance may create undesirable effects. 
So, I think `if type(other) is type(self):` would be more appropriate here.